### PR TITLE
Added reviews with form and respective test suite

### PR DESCRIPTION
**Pull request by amoralescr85**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class ReviewsController < ApplicationController
+
+  def create
+    @review = Review.new(review_params)
+    @show = Show.find(params[:show_id])
+    binding.pry
```
Comment (enthusiastick) [inline app/controllers/reviews_controller.rb]: :point_up: What is this doing here?

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    @review = Review.new
+  end
+
+  private
+
+  def review_params
+    params.require(:review).permit(:title, :description, :rating).merge(
+    show: Show.find(params[:show_id])).merge(
+    user: User.find(current_user)
+                                            )
```
Comment (nwalberts) [inline app/controllers/reviews_controller.rb]: fix this!

File: app/models/review.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Review < ApplicationRecord
+
```
Comment (nwalberts) [inline app/models/review.rb]: fix this

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
   devise_for :users
   root to: "shows#index"
-  resources :shows
   resources :users
+  resources :shows do
```
Comment (nwalberts) [inline config/routes.rb]: Please us the "only" syntax with your routes. I believe you only have actions for index and show at the moment, so only those paths should be available.

File: spec/features/user_creates_review_spec.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+# [ ] I want to get error messages if I do not fill out all required fields
+# [ ] Once I have created an account, I want to be logged in
+
+feature "user creates review" do
+
+  before(:each) do
+    @user = FactoryGirl.create(:user)
+    login_as(@user, scope: :user)
+  end
+
```
Comment (nwalberts) [inline spec/features/user_creates_review_spec.rb]: You create the same show for each of your scenario blocks. As a result, you should actually create it at the top of your test up here with your user.

I would also strongly suggest you use the "let!() syntax over the "before...each...do" syntax, but its not required!

File: spec/features/user_creates_review_spec.rb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    expect(page).to_not have_content("Sign in")
+    expect(page).to_not have_content("Create Account")
+  end
+
+  scenario "fills in bad details in form and submits details" do
+    game_of_thrones = Show.create(
+      title: 'Game of Thrones',
+      description: 'awesome HBO show',
+      average_rating: 5
+                                  )
```
Comment (nwalberts) [inline spec/features/user_creates_review_spec.rb]: fix this

**Comment by harrisonmuskat** [comment]
[As an authenticated user I want to view the details of a tv show so I can review it](https://trello.com/c/XWFhdcTI/31-as-an-authenticated-user-i-want-to-view-the-details-of-a-tv-show-so-i-can-review-it)

**Comment by harrisonmuskat** [comment]
[As an authenticated user I want to view the details of a tv show so I can review it](https://trello.com/c/XWFhdcTI/31-as-an-authenticated-user-i-want-to-view-the-details-of-a-tv-show-so-i-can-review-it)

**Comment by harrisonmuskat** [comment]
[As an authenticated user I want to view the details of a tv show so I can review it](https://trello.com/c/XWFhdcTI/31-as-an-authenticated-user-i-want-to-view-the-details-of-a-tv-show-so-i-can-review-it)

**Comment by harrisonmuskat** [comment]
[As an authenticated user I want to view the details of a tv show so I can review it](https://trello.com/c/XWFhdcTI/31-as-an-authenticated-user-i-want-to-view-the-details-of-a-tv-show-so-i-can-review-it)

**Comment by harrisonmuskat** [comment]
[As an authenticated user I want to add a review to a tv show so that others can see it](https://trello.com/c/qIP1DhXE/24-as-an-authenticated-user-i-want-to-add-a-review-to-a-tv-show-so-that-others-can-see-it)

**Pull request merged**